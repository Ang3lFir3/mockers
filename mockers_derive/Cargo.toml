[package]
authors = ["Mikhail Trishchenkov <kriomant@gmail.com>"]
description = "Macro 1.1 implementation for 'mockers' mocking library"
keywords = ["mock", "test"]
license = "MIT"
name = "mockers_derive"
repository = "https://github.com/kriomant/mockers"
version = "0.10.1"

[dependencies]
syn = { version = "0.11.11", features = ["full"] }
quote = "0.3.15"
itertools = "0.4.13"
lazy_static = "0.2.9"
synom = "0.11.3"

[build-dependencies]

[dev-dependencies]
mockers = { version = "0.10.1", path = "../mockers" }

[lib]
name = "mockers_derive"
proc-macro = true

[features]
debug = []
