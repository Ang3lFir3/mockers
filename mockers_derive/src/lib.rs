#![feature(rustc_macro, rustc_macro_lib)]

extern crate rustc_macro;
extern crate mockers_codegen;

use rustc_macro::TokenStream;

#[rustc_macro_derive(Mock)]
pub fn derive_mock(input: TokenStream) -> TokenStream {
    let item = format!("#[derive(Mock)]\n{}", input);
    let expanded = mockers_codegen::expand_str(&item).unwrap();
    expanded.parse().unwrap()
}
