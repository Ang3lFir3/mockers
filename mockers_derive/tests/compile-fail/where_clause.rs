use mockers_derive::mocked;

#[mocked]
trait A where Self : 'static {}

fn main() {}
