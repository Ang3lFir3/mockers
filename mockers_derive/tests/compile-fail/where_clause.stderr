error: Where clauses are not supported yet.

 --> $DIR/where_clause.rs:4:9
  |
4 | trait A where Self : 'static {}
  |         ^^^^^
